components/channel_header/channel_header.js

```diff
@@ -346,7 +336,24 @@ export default class ChannelHeader extends React.PureComponent {
         }
 
         let headerTextContainer;
-        if (channel.header) {
+        const headerText = (isDirect && dmUser.is_bot) ? dmBot.description : channel.header;
```

Comment:
Surveybot has some extra logic to do this manually. Its description is incorrect though, so I need to submit a PR to change that.